src/pluginLoader.js

```diff
@@ -0,0 +1,660 @@
+      configurable: true
+    });
+
+    var BootstrapPlugin = new BootstrapPluginClass();
+
+    /**
+     * This plugin's name and angular module
+     */
+    BootstrapPlugin.pluginName = "bootstrap-plugin";
+    /**
+     * This plugins logger instance
+     */
+    var log = Logger.get(BootstrapPlugin.pluginName);
+
+    var _module = angular.module(BootstrapPlugin.pluginName, []);
+    _module.config(["$locationProvider", function ($locationProvider) {
+      $locationProvider.html5Mode(true);
+    }]);
+
+    _module.run(['documentBase', function (documentBase) {
+      log.debug("loaded");
+    }]);
+
+    BootstrapPlugin.documentBase = function() {
+      var base = $('head').find('base');
+      var answer = '/'
+      if (base && base.length > 0) {
+        answer = base.attr('href');
+      } else {
+        log.warn("Document is missing a 'base' tag, defaulting to '/'");
+      }
+      return answer;
+    }
+
+    // Holds the document base so plugins can easily
+    // figure out absolute URLs when needed
+    _module.factory('documentBase', function() {
+      return BootstrapPlugin.documentBase();
+    });
+
+    pluginLoader.addModule("ng");
+    pluginLoader.addModule("ngSanitize");
+    pluginLoader.addModule(BootstrapPlugin.pluginName);
+
+    // bootstrap the app
+    $(function () {
+
+      jQuery.uaMatch = function( ua ) {
+        ua = ua.toLowerCase();
+
+        var match = /(chrome)[ \/]([\w.]+)/.exec( ua ) ||
+          /(webkit)[ \/]([\w.]+)/.exec( ua ) ||
+          /(opera)(?:.*version|)[ \/]([\w.]+)/.exec( ua ) ||
+          /(msie) ([\w.]+)/.exec( ua ) ||
+          ua.indexOf("compatible") < 0 && /(mozilla)(?:.*? rv:([\w.]+)|)/.exec( ua ) ||
+          [];
+
+        return {
+          browser: match[ 1 ] || "",
+          version: match[ 2 ] || "0"
+        };
+      };
+
+      // Don't clobber any existing jQuery.browser in case it's different
+      if ( !jQuery.browser ) {
+        var matched = jQuery.uaMatch( navigator.userAgent );
+        var browser = {};
+
+        if ( matched.browser ) {
+          browser[ matched.browser ] = true;
+          browser.version = matched.version;
+        }
+
+        // Chrome is Webkit, but Webkit is also Safari.
+        if ( browser.chrome ) {
+          browser.webkit = true;
+        } else if ( browser.webkit ) {
+          browser.safari = true;
+        }
+
+        jQuery.browser = browser;
+      }
+
+      pluginLoader.loadPlugins(function() {
+
+        if (BootstrapPlugin.injector) {
+          log.debug("Application already bootstrapped");
+          return;
+        }
+
+        var bootstrapEl = pluginLoader.getBootstrapElement();
+        log.debug("Using bootstrap element: ", bootstrapEl);
+
+        BootstrapPlugin._injector = angular.bootstrap(bootstrapEl, pluginLoader.getModules(), {
+          strictDi: false
+        });
+        log.debug("Bootstrapped application");
+      });
+    });
+    return BootstrapPlugin;
+})();
+
```

Comment:
So in the end, should we still name this thing `hawtioPluginLoader`, or leave it globally as `pluginLoader` or `openshiftPluginLoader` && manually update all the references?  Thinking on this a second.  I don't feel strongly either way atm.

Reply:
This was a suggestion from Sam to avoid breaking downstream repos other than our own.

Reply:
My original plan was just to use the global pluginLoader variable and refactor in console and common. We can still do that, but the alias still might be necessary.

Reply:
Works for me then!  @spadgett advice tends to be spot on!

Reply:
Yeah, we documented `hawtioPluginLoader.addModule` as the way to add modules, so we have to keep that working to avoid breaking customer extensions. I'm OK renaming and fixing up our code as long as we keep it as an alias (with a comment explaining).

https://docs.openshift.org/latest/install_config/web_console_customization.html

Reply:
I'll add a comment to the LOC where I create the alias. I can also refactor to use 'pluginLoader' in our repos if we want to do that.